test/e2e/flowaggregator_test.go

```diff
@@ -754,10 +790,21 @@ func getUnit64FieldFromRecord(t *testing.T, record string, field string) uint64
 // received all the expected records for a given flow with source IP, destination IP
 // and source port. We send source port to ignore the control flows during the
 // iperf test.
-func getCollectorOutput(t *testing.T, srcIP, dstIP, srcPort string, timeStart time.Time, checkAllRecords bool) (string, []string) {
+func getCollectorOutput(t *testing.T, srcIP, dstIP, srcPort string, timeStart time.Time, checkAllRecords bool, restartInfo restartInfo) (string, []string) {
 	var collectorOutput string
 	var recordSlices []string
-	err := wait.PollImmediate(500*time.Millisecond, aggregatorInactiveFlowRecordTimeout, func() (bool, error) {
+	timeout := aggregatorInactiveFlowRecordTimeout
+	if restartInfo.isRestart {
+		timeout = restartFlowAggregatorTimeout
+	}
+	err := wait.PollImmediate(500*time.Millisecond, timeout, func() (bool, error) {
+		// When checking restart flows, we need to create traffic until one flow record
+		// is received correctly since connection re-establish between Flow Exporter and
+		// Flow Aggregator will happen any time during connCheckInterval.
+		if restartInfo.isRestart {
+			stdout, stderr, err := restartInfo.data.runCommandFromPod(testNamespace, restartInfo.sourcePodName, busyboxContainerName, strings.Fields(restartInfo.cmd))
+			require.NoErrorf(t, err, "Error when running wget command, stdout: %s, stderr: %s", stdout, stderr)
+		}
```

Comment:
Thanks for addressing this in a different way. Instead of adding an argument to `getCollectorOutput` function, I think creating a separate function with an appropriate name indicating repeated test flows is a cleaner approach. Please consider that.

Reply:
Technically we can create the traffic at a certain period other than 500ms based on flow aggregator inactive timeout. If this works, we can keep this as it is.

Reply:
Created a new function to handle restart case. Since flow aggregator inactive timeout is 6s and checkConnInterval is 10s currently, it will cause some flakiness when we only send record at 6s, 12s because the second record may not be received if total wait time is 16s. Set wait interval to 2s instead. Thanks!